lib/b2b/passwords.ts

```diff
@@ -0,0 +1,277 @@
+  member_session?: MemberSession;
+  organization: Organization;
+}
+
+export interface EmailResetStartRequest {
+  organization_id: string;
+  email_address: string;
+  login_redirect_url?: string;
+  reset_password_redirect_url?: string;
+  reset_password_expiration_minutes?: number;
+  reset_password_template_id?: string;
+  code_challenge?: string;
+  locale?: string;
+}
+
+export interface EmailResetStartResponse extends BaseResponse {
+  member_id: string;
+  member_email_id: string;
+}
+
+export interface EmailResetRequest {
+  password_reset_token: string;
+  password: string;
+  session_token?: string;
+  session_jwt?: string;
+  session_duration_minutes?: number;
+  session_custom_claims?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  code_verifier?: string;
+}
+
+export interface EmailResetResponse extends BaseResponse {
+  member_id: string;
+  member_email_id: string;
+  organization_id: string;
+  member: Member;
+  session_token?: string;
+  session_jwt?: string;
+  member_session?: MemberSession;
+  organization: Organization;
+}
+
+export interface ExistingPasswordResetRequest {
+  email_address: string;
+  existing_password: string;
+  new_password: string;
+  session_token?: string;
+  session_jwt?: string;
+  session_duration_minutes?: number;
+  session_custom_claims?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  organization_id: string;
+}
+
+export interface ExistingPasswordResetResponse extends BaseResponse {
+  member_id: string;
+  organization: Organization;
+  member: Member;
+  session_token?: string;
+  session_jwt?: string;
+  member_session?: MemberSession;
+}
+
+export interface SessionResetRequest {
+  password: string;
+  organization_id: string
+  session_token?: string;
+  session_jwt?: string;
+}
+
+export interface SessionResetResponse extends BaseResponse {
+  member_id: string;
+  organization: Organization;
+  member: Member;
+  member_session?: MemberSession;
+}
+
+export interface StrengthCheckRequest {
+  email_address?: string;
+  password: string;
+}
+
+export interface StrengthCheckResponse extends BaseResponse {
+  valid_password: boolean;
+  score: number;
+  breached_password: boolean;
+  strength_policy: string;
+  breach_detection_on_create: boolean;
+  zxcvbn_feedback: {
+    suggestions: string[];
+    warning: string;
+  }
+  luds_feedback: {
+    has_lower_case: boolean;
+    has_upper_case: boolean;
+    has_digit: boolean;
+    has_symbol: boolean;
+    missing_complexity: number;
+    missing_characters: number;
+  };
+}
+
+interface MigrateRequestBase {
+  organization_id: string;
+  email_address: string;
+  hash: string;
+  name?: string;
+  trusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  untrusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+}
+
+interface MD5MigrateRequest extends MigrateRequestBase {
```

Comment:
Can we reuse the same migrate request definitions from B2C?
Something like
```
// Shared
interface MD5MigrateRequestShared  {
      hash_type: "md_5";
    md_5_config?: {
    prepend_salt?: string;
    append_salt?: string;
  };  
}

// Unique per vertical
interface MigrateRequestBase {
  organization_id: string;
  email_address: string;
  hash: string;
  name?: string;
  trusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
  untrusted_metadata?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
}

type MD5MigrateRequest = MD5MigrateRequestShared & MigrateRequestBase;
```

Reply:
yeah probably lemme give it a go

Reply:
https://github.com/stytchauth/stytch-node/pull/206/commits/dce0ad2ac6fdd3af946b253052c477a859f2ef81
Does that work? lmk if it can be even cleaner

Reply:
Looks great!